docs/development.rst

```diff
@@ -10,6 +10,40 @@ This chapter will get you started with |project_name| development.
 |project_name| is written in Python (with a little bit of Cython and C for
 the performance critical parts).
 
+Contributions
+-------------
+
+... are welcome!
+
+Some guidance for contributors:
+
+- discuss about changes on github issue tracker, IRC or mailing list
+
+- choose the branch you base your changesets on wisely:
+
+  - choose x.y-maint for stuff that should go into next x.y release
+    (it usually gets merged into master branch later also)
+  - choose master if that does not apply
+
+- do clean changesets:
+
+  - focus on some topic, resist changing anything else.
+  - do not do style changes mixed with functional changes.
+  - if you need to fix something after commit, do a fixup commit you can
+    merge into the bad commit later.
```

Comment:
if no review is ongoing, editing the commit and fixing it directly is ofc also ok.
what i primarily meant here is not to merge the fix into some arbitrary unrelated commit.